src/main/java/com/google/devtools/build/lib/skyframe/PackageFunction.java

```diff
@@ -1266,7 +1272,20 @@ private CompiledBuildFile compileBuildFile(
       // If control flow reaches here, we're in territory that is deliberately unsound.
       // See the javadoc for ActionOnIOExceptionReadingBuildFile.
     }
-    ParserInput input = ParserInput.fromLatin1(buildFileBytes, inputFile.toString());
+    StoredEventHandler handler = new StoredEventHandler();
+    Optional<ParserInput> input =
+        SkyframeUtil.createParserInput(
+            buildFileBytes,
+            inputFile.toString(),
+            semantics.get(BuildLanguageOptions.INCOMPATIBLE_ENFORCE_STARLARK_UTF8),
+            handler);
+    if (input.isEmpty()) {
+      return new CompiledBuildFile(
```

Comment:
CompiledBuildFile doesn't make sense here: we failed before even attempting to compile. There is no trace of an AST.

Instead, I'd throw a PackageFunctionException, something like
```java
    throw PackageFunctionException.builder()
        .setType(PackageFunctionException.Type.BUILD_FILE_CONTAINS_ERRORS)
        .setPackageIdentifier(packageId)
        .setTransience(Transience.PERSISTENT)
        .setException(...)
        .setMessage("error reading " + inputFile.toString())
        .setPackageLoadingCode(PackageLoading.Code.STARLARK_EVAL_ERROR)
        .build();
````